File: IT/Eccezioni/Esercizi/Python/esercizio5.py
# file esercizio5.py

from __future__ import print_function

nums = ["1", "2", "", "3", "abc", "7", "def"]
print(nums)

tot = 0

for n in nums:
	try:
		tot += int(n)
	except ValueError as e:
		print(e)

print("Somma dei numeri presenti: {}".format(tot))
print("Fine del programma.")

File: IT/1/Eccezioni/Esercizi/Java/Esercizio5.java
/* file Esercizio5.java */

import java.util.Arrays;

public class Esercizio5 {
	public static void main(String [] args) {
		String [] nums = {"1", "2", "", "3", "abc", "7", "def"};

		int tot = 0;

		for (String n : nums) {
			try {
				tot += Integer.parseInt(n);
			}
			catch (NumberFormatException e) {
				System.out.println(e.getMessage());
			}
		}

		System.out.println("Somma dei numeri presenti: " + tot);
		System.out.println("Fine del programma.");
	}
}

